### DISCUSSION TITLE
Support for Intel GPU

### DISCUSSION THREAD

**Discussion by koirasp** [mcmonkeyprojects/SwarmUI#474] [General]
Hi! I use a Surface Pro 9 computer. May you please enable Swarm to use Intel GPUs such as mine?

**Comment by mcmonkey4eva** [comment]
From a check at google, that has an Intel integrated legacy iris xe gpu, not a modern intel arc gpu. So... probably not gonna work.

It's maybe possible, I don't entirely know tbh, IPEX I think technically supports Iris. So, see docs for the comfy backend @ https://github.com/comfyanonymous/ComfyUI?tab=readme-ov-file#intel-gpus

Swarm has a standard comfy install as it's backend, so I think you'd just need the ipex prereqs installed.

If it *does* work, it's going to be horrendously slow, because you have the lowest tier of Intel's previous generation of integrated graphics.
(1) lowest tier mobile chip = slow
(2) intel = slow
(3) previous gen = slow
(4) integrated = slow

slow to the fourth power yields barely usable if it works at all, so, uh... good luck?

If you can afford buying things, even an entry level desktop PC with a previous gen low tier nvidia chip would at least be usable, an RTX 3060 is really solid (though obviously much slower than a 4090). Or there are various web services that let you run a PC with a GPU for like a buck per hour (eg runpod, massedcompute, ...) and you can easily use Swarm on those.

**Reply by koirasp** [reply]
OMG I'm so sad now 😭 I thought I have bought a badass computer

**Comment by Griphen116** [comment]
@mcmonkey4eva given that the new intel GPUs that are releasing show a lot of potential for a good price, and the last gen cards are still not bad at all, still might be nice to have that support natively in Swarm to open up to more users.

**Reply by mcmonkey4eva** [reply]
Yes. I'd love a PR from someone who has an intel gpu and knows what they're doing to add that
